src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -94,27 +75,164 @@ private void WriteStatisticsToFile(CSVMetricReader reader)
+            {
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "Test Name" }, Alignment = TableColumnAlignment.Left },
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "Metric" }, Alignment = TableColumnAlignment.Left },
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "Iterations" }, Alignment = TableColumnAlignment.Center },
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "AVERAGE" }, Alignment = TableColumnAlignment.Right },
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "STDEV.S" }, Alignment = TableColumnAlignment.Right },
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "MIN" }, Alignment = TableColumnAlignment.Right },
+                new TableColumn(){ HeaderCell = new TableCell() { Text = "MAX" }, Alignment = TableColumnAlignment.Right },
+            };
+
+            Console.WriteLine(mdTable);
+
+            // TODO: Save table as Markdown file!
+        }
+
+        /// <summary>
+        /// Generate CSV data (Probably not the most efficient way, e.g. How big can this become).
+        /// </summary>
+        /// <param name="reader"></param>
+        private static void WriteStatisticsToFile(string configurationRunId, CSVMetricReader reader)
+        {
+            var statisticsTable = CreateStatistics(reader);
+
+            // Only create output statistics if there is data.
+            if (statisticsTable.Rows.Count() > 0)
+            {
+                PrintStatistics(statisticsTable);
+
+                var currentWorkingDirectory = Directory.GetCurrentDirectory();
+                var statisticsFilePath = Path.Combine(currentWorkingDirectory, $"{configurationRunId}-Statistics.csv");
+                statisticsTable.WriteToCSV(statisticsFilePath);
+                WriteInfoLine($"Statistics written to \"{statisticsFilePath}\"");
+            }
+        }
+
+        #region IDisposable implementation
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        ~XunitPerformanceHarness()
+        {
+            Dispose(false);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                    FreeManagedResources();
+                FreeUnManagedResources();
+                _disposed = true;
+            }
         }
+
+        private void FreeManagedResources()
+        {
+            // Close the log when all test cases have completed execution.
+            // HACK: This is a hack because we haven't found a way to close the file from within xunit.
+            BenchmarkEventSource.Log.Close();
+
+            // Process the results now that we know we're done executing tests.
+            ProcessResults();
+        }
+
+        private void FreeUnManagedResources()
```

Comment:
No, I will remove it.

---
In reply to: [100352500](https://github.com/Microsoft/xunit-performance/pull/183#discussion_r100352500) [](ancestors = 100352500)